### PR TITLE
MM-57112 Add new sources to push proxy server logs

### PR DESCRIPTION
**Pull request by sri-byte**
Impact:
Add new source tables to push proxy server logs. 

- [x] Add new tables to staging sources.
- [x] Create staging models.
- [x] Create intermediate models.
- [x] Add new intermediate models to mart `fct` model.
- [x] Update yml files

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-57112

### CODE REVIEW

File: transform/mattermost-analytics/models/staging/push_proxy/_push_proxy__sources.yml [hunk 1]
```diff
@@ -312,3 +312,309 @@ sources:
               The actions taken when processing the request, enclosed in double quotes. This value is a comma-separated
               list that can include the values described in Actions taken. If no action was taken, such as for 
               a malformed request, this value is set to -.
+
+      - name: logs_eu_new
+        description: Logs from test notification server.
```
Comment (catalintomai) [inline transform/mattermost-analytics/models/staging/push_proxy/_push_proxy__sources.yml:317]: why 'test'?

File: transform/mattermost-analytics/models/staging/push_proxy/_push_proxy__sources.yml [hunk 2]
```diff
@@ -312,3 +312,309 @@ sources:
...
+              a malformed request, this value is set to -.
+          - name: redirect_url
+          - name: error_reason
+          - name: target_port_list
+          - name: target_status_code_list
+          - name: classification
+          - name: classification_reason
+
+      - name: logs_us_new
+        description: Logs from test notification server.
```
Comment (catalintomai) [inline transform/mattermost-analytics/models/staging/push_proxy/_push_proxy__sources.yml:521]: why 'test'?

File: transform/mattermost-analytics/models/staging/push_proxy/_push_proxy__sources.yml (hunk 1 above)
Comment (sri-byte) [reply transform/mattermost-analytics/models/staging/push_proxy/_push_proxy__sources.yml:317]: Updated, was a copy paste oversight.

File: transform/mattermost-analytics/models/staging/push_proxy/_push_proxy__sources.yml (hunk 2 above)
Comment (sri-byte) [reply transform/mattermost-analytics/models/staging/push_proxy/_push_proxy__sources.yml:521]: updated

**Review by ifoukarakis — CHANGES_REQUESTED** [review]
Value addition must handle nulls.

Some small issues in docs as well.

File: transform/mattermost-analytics/models/marts/product/notifications/fct_notification_stats.sql [hunk 3]
```diff
@@ -1,13 +1,17 @@
 select
     spine.date_hour,
-    coalesce(eu_logs.count_send_push, 0) as count_eu_send_push,
-    coalesce(eu_logs.count_send_ack, 0) as count_eu_send_ack,
-    coalesce(us_logs.count_send_push, 0) as count_us_send_push,
-    coalesce(us_logs.count_send_ack, 0) as count_us_send_ack,
-    coalesce(test_logs.count_send_push, 0) as count_test_send_push,
-    coalesce(test_logs.count_send_ack, 0) as count_test_send_ack
+    coalesce(eu_logs.count_send_push + logs_eu.count_send_push, 0) as count_eu_send_push,
```
[the remark this answers is not in the document]
Comment (sri-byte) [reply transform/mattermost-analytics/models/marts/product/notifications/fct_notification_stats.sql]: Updated, used `dbt_utils.safe_add`.

File: transform/mattermost-analytics/models/staging/push_proxy/_push_proxy__sources.yml [hunk 4]
```diff
@@ -312,3 +312,309 @@ sources:
...
+            description: |
+              The actions taken when processing the request, enclosed in double quotes. This value is a comma-separated
+              list that can include the values described in Actions taken. If no action was taken, such as for 
+              a malformed request, this value is set to -.
+          - name: redirect_url
+          - name: error_reason
+          - name: target_port_list
+          - name: target_status_code_list
+          - name: classification
+          - name: classification_reason
```
[the remark this answers is not in the document]
Comment (sri-byte) [reply transform/mattermost-analytics/models/staging/push_proxy/_push_proxy__sources.yml]: Added descriptions.

**Review by ifoukarakis — COMMENTED** [review]
The `test` mentions persist. Same for missing docs.

**Comment by sri-byte** [comment]
Thanks @ifoukarakis, I missed updating the staging model.yml, had to update the descriptions in 2 places.

**Review by ifoukarakis — APPROVED** [review]

**Pull request merged**